extensions/mongodb-client/runtime/pom.xml

```diff
@@ -29,11 +30,25 @@
         </dependency>
         <dependency>
             <groupId>org.mongodb</groupId>
-            <artifactId>mongo-java-driver</artifactId>
+            <artifactId>mongodb-driver-sync</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.mongodb</groupId>
+            <artifactId>mongodb-driver-async</artifactId>
         </dependency>
         <dependency>
             <groupId>org.mongodb</groupId>
             <artifactId>mongodb-driver-reactivestreams</artifactId>
+            <exclusions>
+                <!--
+                    Exclude mongodb-driver-async as it shares dependencies with mongodb-driver-sync so we need
+                    to have both versions aligned which is not the case if it is a transitive dependency of mongodb-driver-reactivestreams.
```

Comment:
If you declare the `mongo-driver-async` in the BOM I think you can remove this exclusion

Reply:
Yes, this was again a leftover of my attemp to fix the dependencies issue